### PR TITLE
Victor/825 pagination

### PR DESCRIPTION
**Pull request by victordelval**
## Description

Closes: #825

- The pagination component
- Hook for component states and paginated query parameters. For convenience it also returns the component itself with no properties.
- Custom styles and subcomponents (next and prev buttons have been adapted to include the "dark" flag and display them in the green style)
- Parameterizable credit batches query
- Component Storybook

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles 
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-components/src/components/table/useTablePagination.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    setRowsPerPage(parseInt(event.target.value as string, 10));
+    setOffset(0);
+    setPage(0);
+  };
+
+  let paginationProps = {
```
Comment (haveanicedavid) [inline web-components/src/components/table/useTablePagination.ts:51]: this is just personal preference, but I prefer using `const` anytime a variable isn't going to be reassigned

File: web-registry/src/components/organisms/CreditBatches.tsx [hunk 2]
```diff
@@ -78,42 +79,58 @@ const useStyles = makeStyles(theme => ({
   grey: {
     color: theme.palette.info.main,
   },
+  tableBody: props => ({
```
Comment (haveanicedavid) [inline web-registry/src/components/organisms/CreditBatches.tsx:82]:
```suggestion
  tableBody: {
```

**Review by haveanicedavid — APPROVED** [review]
A few comments, but it's looking really good! Great work Victor!

File: web-components/src/components/table/useTablePagination.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ * data in the source.
+ */
+export function useTablePagination(rowsPerPageOptions: number[] = [5]): UseTablePaginationResult {
+  const [page, setPage] = useState<number>(0);
+  const [rowsPerPage, setRowsPerPage] = useState<number>(rowsPerPageOptions[0]);
+  const [offset, setOffset] = useState<number>(0);
+  const [countTotal, setCountTotal] = useState<number>(0);
+  const [paginationParams, setPaginationParams] = useState<URLSearchParams | null>(null);
+
+  const prepareParams = (rowsPerPage: number, offset: number): void => {
```
Comment (mhagel) [inline web-components/src/components/table/useTablePagination.ts:24]: nicely done

File: web-components/src/components/table/useTablePagination.ts (hunk 1 above)
Comment (mhagel) [reply web-components/src/components/table/useTablePagination.ts:51]: agree

**Review by mhagel — APPROVED** [review]
Tested. Works and looks really great. Well done!

**Pull request merged**